hudi-common/src/main/java/org/apache/hudi/avro/HoodieAvroReaderContext.java

```diff
@@ -69,15 +70,26 @@
  */
 public class HoodieAvroReaderContext extends HoodieReaderContext<IndexedRecord> {
   private final String payloadClass;
+  private final Map<StoragePath, HoodieAvroFileReader> reusableFileReaders;
```

Comment:
This is an optional cache of readers, this is only used by the HoodieBackedTableMetadata right now

Reply:
Wow, we even support reusing from multiple table paths now?

Reply:
Even today you can read from multiple table paths with the same context technically. In practice though, we do not.